utils/handlers/pubsub.go

```diff
@@ -36,7 +36,7 @@ func HandlePublishCall(pubsub *pubsub.Module, auth *auth.Module) http.HandlerFun
 		}
 		token := strings.TrimPrefix(tokens[0], "Bearer ")
 
-		status, err := pubsub.Publish(project, token, subject, b)
+		status, err := pubsub.Publish(project, token, req.Subject, []byte(req.Data))
 
```

Comment:
Dont take a byte array as a parameter, take an interface. You should marshal inside.

Reply:
Parse an interface directly here. No need to marshal again inside the Publish() function